### DISCUSSION TITLE
Using links in translation values

### DISCUSSION THREAD

**Discussion by ClementDreptin** [amannn/next-intl#587] [Q&A] [answered]
Hi,

I've been trying to use links in translation values but I don't how to go about setting the `<a>` attributes. I tried to do something like this
`translations.json`
```JSON
{
  "content": "Go to <a href=\"https://mywebsite.com/\">my website</a>"
}
```
`component.tsx`
```TSX
function Component() {
  const t = useTranslations();

  return <p>{t.rich("content", { a: (chunks) => <a>{chunks}</a> })}</p>;
}
```
but I get an `INVALID_TAG` error.

I found a dirty and possibly dangerous way of doing it by inserting the translation value as raw HTML
```TSX
function Component() {
  const t = useTranslations();

  return <p dangerouslySetInnerHTML={{ __html: t.raw("content") }} />;
}
```
but I'd rather not doing that if I can.

I was naively thinking this would be a common issue but I can't find anything online, maybe I'm searching wrong. Has anyone done this before, if so, how?

Thanks in advance

**Comment by amannn** [comment, marked as answer]
Thank you for the question!

I've just added an expandable section "How can I pass attributes to tags?" to [the rich text docs](https://next-intl-docs.vercel.app/docs/usage/messages#rich-text). Hope that helps! 🙌

**Reply by ClementDreptin** [reply]
Thanks you for your answer!

I ended up creating a separate tag for each link like so
`translations.json`
```JSON
{
  "content": "Buy some <shoesLink>shoes</shoesLink> and <shirtsLink>shirts</shirtsLink>"
}
```
`component.tsx`
```TSX
function Component() {
  const t = useTranslations();

  return (
    <p>
      {t.rich("content", {
        shoesLink: (chunks) => <a href="https://mywebsite.com/shoes">{chunks}</a>,
        shirtsLink: (chunks) => <a href="https://mywebsite.com/shirts">{chunks}</a>
      })}
    </p>
  );
}
```
I thought doing that was easier than creating an extra message for each link. I'm not 100% satisfied with this solution and really wish we could actually pass attributes to tags in the messages but I know the rich text format is out of your control.
